### ISSUE TITLE
Automate accessibility checks into build/pipeline

### ISSUE THREAD

**Issue by jonhurrell** [dwp/design-system#82]
## Context
We need to automate a11y testing for common problems when developing/release to speed up the manual test and to give greater assurance. This could be done using axe-core and PA11Y and Selenium.

This will help to meet the following needs of our users:

- I can see that things are accessible (high level need)
- best (accessible) code examples
- built-in accessibility

## Alternatives
Manual testing with browser plugins, following advice and guidance from https://accessibility-manual.dwp.gov.uk/best-practice/manual-testing-using-tools/#accessibility-insights-assessment-extension

## Additional information (if applicable)

- https://accessibility-manual.dwp.gov.uk/best-practice/automated-testing-using-axe-core-and-pa11y
- See the discussion below

**Comment by abbott567** [comment]
It's primitive, but I've got some basic tests using Selenium Webdriver, AxeCore, Mocha and Chai to test for accessibility issues and broken links which might be useful. https://github.com/dwp/accessibility-manual/tree/master/test

![image](https://user-images.githubusercontent.com/9450064/102122466-6ac2a380-3e3d-11eb-8a9a-f073118efd99.png)

**Comment by simoneduca** [comment]
I've edited this issue title in light of Gitlab support for running pa11y in the pipeline. As [Craig points out](https://github.com/abbott567/axe-core-vs-pa11y) though, pa11y alone isn't enough and we need to figure out if we can ran both pa11y and axe-core to have enough coverage of potential issues.

**Comment by abbott567** [comment]
Unfortunately on more digging, it looks like GitLab's PA11Y can't run axe-core at the moment as it runs as 'zero config' and you need to be able to configure the runners or it only uses HTMLCS by default. 

I've updated the [Selenium tests in the accessibility manual](https://github.com/dwp/accessibility-manual/blob/master/test/acceptance/common-tests/axe-core.js) so they run in parallel now and it runs 10 to 12 tests at a time which has cut the acceptance testing the whole application down from around 15 minutes to about 2. 

You can also run PA11Y from Mocha and pass axe-core in as a runner relatively easily if Selenium is overkill, but it doesn't look like relying on GitLab is a viable option at the moment.

**Comment by simoneduca** [comment]
MR is ready to review:
- generates pa11y html reports locally with npm run reports
- adds a pa11y-ci pipeline job, which runs on the preview urls (waits till deploy job has finished)
- in the pipeline, the html reports are attached to the pa11y-ci job (under Job artifacts>browse)
- the job is allowed to fail to allow more flexibility
